UPSTREAM_PROJECTS.yaml

```diff
@@ -153,6 +153,8 @@ projects:
             go_version: "1.21"
           - tag: cluster-autoscaler-1.30.1
             go_version: "1.22"
+          - tag: cluster-autoscaler-1.30.1
```

Comment:
Can we use the [cluster-autoscaler-1.31.0](https://github.com/kubernetes/autoscaler/releases/tag/cluster-autoscaler-1.31.0) tag here?

Reply:
I tried using that but the patches didn't apply cleanly. I was able to manually apply the first two patches but the third patch to update go.mod dependencies didn't work and it doesn't look straighforward to update that

Reply:
I will upgrade to 1.31 in a separate PR as I certainly need to update the patches especially the third patch where we remove go.mod dependencies. I tried running go mod tidy in the upstream repo checked out with 1.31 tag after applying the first 2 patches but it didn't remove any go dependencies. I will have to check [changelog](https://github.com/kubernetes/autoscaler/releases/tag/cluster-autoscaler-1.31.0) for 1.31 and update the patches accordingly.